Add basic concept and glossary entries (#14402)
trinodb/trino: docs/src/main/sphinx/overview/concepts.rst

    provide further information about Trino and the concepts in use.

> .. _trino-concept-architecture:

Reviewer: Is there an advantage to adding an anchor if we're not going to use it?

I think there's a few others like this.

Author: Maybe it's preemptive but I think making the standard that we add these when defining glossary entries or concepts would be good, that way it's not a guessing game of if a term has a link already set up. Plus, doing this as needed will invite inconsistency (which already exists). One PR I want to do is creating a consistent naming standard for these like `<page-title>-<term/concept/property/header>`.

Reviewer: I don't think it's much of a guessing game as to whether or not an anchor exists (you can just look), and I think the strategy of "when needed" is as consistent as it needs to be.

I think the litmus test to use is whether or not taking the time to proactively make everything consistent would make things easier/better to work with. I don't really see any meaningful upside of adding a bunch of anchors that don't do anything - it'd just be a waste of time. (I _am_ ok with trying to standardize naming, though.)

Architecture
------------

Trino is a distributed query engine. Trino processes data in parallel across

> multiple servers. There are two types of Trino servers:

Reviewer 2: Condense into one sentence with
> distributed query engine that processes data in parallel

> multiple servers. There are two types of Trino servers:

Reviewer 2: Replace colon with comma

:ref:`coordinators <trino-concept-coordinator>` and
:ref:`workers <trino-concept-worker>`. The following section explains the the

> difference between the two and how the cluster processes data.

Reviewer 2: Reword to
> The following sections describe these servers and other components of Trino's architecture.

.. _trino-concept-cluster:

Cluster
^^^^^^^

A Trino cluster consists of a :ref:`coordinator <trino-concept-coordinator>` and
many :ref:`workers <trino-concept-worker>`. Users connect to the coordinator
with their :ref:`SQL <glossSQL>` query tool. The coordinator collaborates with the

> workers. The coordinator as well as all the workers access the connected

Reviewer: ```suggestion
workers. The coordinator and all the workers access the connected
```
